pkg/knowledge_base/edge_kb.go

```diff
@@ -78,6 +79,20 @@ const (
 	Downstream Reuse = "downstream"
 )
 
+func MergeKBs(kbsToUse []EdgeKB) (EdgeKB, error) {
+	kb := EdgeKB{}
+	var err error
+	for _, currKb := range kbsToUse {
+		for edge, detail := range currKb {
+			if _, found := kb[edge]; found {
+				err = errors.Join(err, fmt.Errorf("edge for %s -> %s is already defined in the knowledge base", edge.Source, edge.Destination))
```

Comment:
Do `EdgeKB` have a name? Would be nice when we have more than just 2 to have that in the error message.

Reply:
yeah ill add to error message

Reply:
they dont have names actually

Reply:
Yeah, I figured - when we add the next one (so there's > 2) we should add a name so we can get that in the error.